generator/wrapper_gen.py

```diff
@@ -245,6 +251,240 @@ def daal_generate_shuffled_indices(idx, random_state):
+                tree_id=tree_id, feature_index=struct["split_feature"],
+                feature_value=struct["threshold"],
+                parent_id=parent_id, position=position)
+
+            # append children
+            node_stack.append(Node(struct["left_child"], parent_id, 0))
+            node_stack.append(Node(struct["right_child"], parent_id, 1))
+
+    return mb.model()
+
+
+def get_gbt_model_from_xgboost(booster: Any) -> Any:
+    class Node:
+        def __init__(self, tree: str, parent_id: int, position: int):
+            self.tree = tree
+            self.parent_id = parent_id
+            self.position = position
+
+    booster.dump_model("raw.txt")
+    with open("raw.txt", "r") as read_file:
+        xgb_model = read_file.read()
+    remove("./raw.txt")
+
+    xgb_config = json.loads(booster.save_config())
+
+    updater = list(xgb_config["learner"]["gradient_booster"]["updater"].keys())[0]
+    max_depth = int(xgb_config["learner"]["gradient_booster"]
+                    ["updater"][updater]["train_param"]["max_depth"])
+    n_features = int(xgb_config["learner"]["learner_model_param"]["num_feature"])
+    n_classes = int(xgb_config["learner"]["learner_model_param"]["num_class"])
+    base_score = float(xgb_config["learner"]["learner_model_param"]["base_score"])
+
+    is_regression = False
+    objective_fun = xgb_config["learner"]["learner_train_param"]["objective"]
+    if n_classes > 2:
+        if objective_fun in ["multi:softprob", "multi:softmax"]:
+            print("Found multiclass classification")
+        else:
+            raise TypeError(
+                "multi:softprob and multi:softmax are only supported for multiclass classification")
+    elif objective_fun.find("binary:") == 0:
+        if objective_fun in ["binary:logistic", "binary:logitraw"]:
+            print("Found binary classification")
+            n_classes = 2
+        else:
+            raise TypeError(
+                "binary:logistic and binary:logitraw are only supported for binary classification")
+    else:
+        print("Found regression. If it's not, your objective is not supported")
+        is_regression = True
+
+    last_booster_start = xgb_model.rfind("booster[") + 8
+    n_iterations = int((int(xgb_model[last_booster_start:xgb_model.find(
+        "]", last_booster_start)]) + 1) / (n_classes if n_classes > 2 else 1))
+
+    # Create + base iteration
+    if is_regression:
+        mb = gbt_reg_model_builder(n_features=n_features, n_iterations=n_iterations + 1)
+
+        tree_id = mb.create_tree(1)
+        mb.add_leaf(tree_id=tree_id, response=base_score)
+    else:
+        mb = gbt_clf_model_builder(
+            n_features=n_features, n_iterations=n_iterations, n_classes=n_classes)
+
+    class_label = 0
+    iterations_counter = 0
+    for tree in xgb_model.expandtabs(1).replace(" ", "").split("booster[")[1:]:
+        if is_regression:
+            tree_id = mb.create_tree(len(tree.split("\\n")) - 1)
```

Comment:
why we have -1 here?

Reply:
beacuse there is one extra-\n in `tree` after split.
It looks like:
\n
node\n
...
node\n

Reply:
I made a mistake. There's no extra-\n at the start of a `tree`, but python's `split()` method also returns the empty lines, so here it works like:
`tree == r"""
node1\n
node2\n
...
noden\n"""`
and `tree.split('\n')` will return `['node1', 'node2', ..., 'noden', '']`, which length is one greater than actual node count.